cmd/cloud-controller-manager/main.go

```diff
@@ -59,18 +62,19 @@ func main() {
 	nodeIpamController := nodeIPAMController{}
 	nodeIpamController.nodeIPAMControllerOptions.NodeIPAMControllerConfiguration = &nodeIpamController.nodeIPAMControllerConfiguration
 	fss := cliflag.NamedFlagSets{}
-	nodeIpamController.nodeIPAMControllerOptions.AddFlags(fss.FlagSet("nodeipam controller"))
+	nodeIpamController.nodeIPAMControllerOptions.AddFlags(fss.FlagSet(kcmnames.NodeIpamControllerName))
 
-	controllerInitializers["nodeipam"] = app.ControllerInitFuncConstructor{
+	controllerInitializers[kcmnames.NodeIpamControllerName] = app.ControllerInitFuncConstructor{
 		// "node-controller" is the shared identity of all node controllers, including node, node lifecycle, and node ipam.
 		// See https://github.com/kubernetes/kubernetes/pull/72764#issuecomment-453300990 for more context.
 		InitContext: app.ControllerInitContext{
 			ClientName: "node-controller",
 		},
 		Constructor: nodeIpamController.StartNodeIpamControllerWrapper,
 	}
+	controllerAliases["nodeipam"] = kcmnames.NodeIpamControllerName
```

Comment:
I would like to have the controller name  defined in the same place/module where the controller lives. And the nodeipam controller lives in KCM and is imported by CCM

https://github.com/kubernetes/kubernetes/blob/894cfdfeb4a09083849f9193f7c36360e61509eb/cmd/cloud-controller-manager/nodeipamcontroller.go#L37-L39

KCM imports CCM controllers/options/names as well. We prevent circular dependency by choosing the relevant packages for nodeipam controller and only importing them in the `main` package of CCM

https://github.com/kubernetes/kubernetes/blob/894cfdfeb4a09083849f9193f7c36360e61509eb/cmd/cloud-controller-manager/.import-restrictions#L4

Reply:
I thought about this in a way that the old name belonged to CCM when the aliases did not exist yet. Theoretically it could have been a different name from KCM.
 
- due to import restrictions I am not defining it in CCMControllerAliases
- all the nodeipam controller relevant bits are in CCM `main` package so it is in a relevant place

I suppose I could also import and use the KCMControllerAliases, if we don't want to have a lonely string around.